### PR TITLE
mavsys: mode: add solutions for setting AUTO.MISSION and AUTO.LOITER modes

### PR DESCRIPTION
**Pull request by TSC21**
Also the enums are updated - can push this to another PR if needed.

### CODE REVIEW

File: mavros/scripts/mavsys [hunk 1]
```diff
@@ -20,7 +20,15 @@ from mavros_msgs.srv import SetMode, StreamRate, StreamRateRequest
 
 
 def do_mode(args):
+<<<<<<< HEAD
     base_mode = 0
+=======
```
Comment (khancyr) [inline mavros/scripts/mavsys]: looks like rebase error ^^

Comment (TSC21) [reply mavros/scripts/mavsys]: That's fixed @khancyr. thanks

**Comment by TSC21** [comment]
Looks like we don't need to unset the datalink failsafe. For PX4, the heartbeat type published by `sys_status`, for now, needs to be of type `MAV_TYPE_GCS` so to be accepted and published on the `telemetry_status` uORB msg and that way set the datalink stream. IOT to it be accepted also with `MAV_TYPE_ONBOARD_CONTROLLER`, I will purpose a patch to the Firmware.
And that is not all: https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_receiver.cpp#L1718 - the 1st condition is that `msg->sysid != mavlink_system.sysid`, so we cannot have `system_id == tgt_system_id`. And that's why I'm adding that option by default to the launcher filers.

**Comment by TSC21** [comment]
@shakthi-prashanth-m for your interest.

**Comment by TSC21** [comment]
@khancyr in Ardupilot is there any special case like this? Do you treat the hearbeat differently according to the `MAV_TYPE`?

**Comment by vooon** [comment]
Mavros sysid may be set via url, addition to launch is not required.
But for me that check looks weird, because OBC is part of same Unmanned System as FCU.

So FCU should understand when (hb.sysid == my.sysid && hb.type == OBC) - we have link with our OBC.

**Pull request merged**